### PR TITLE
Copy / Delete Sheet

### PR DESCRIPTION
**Pull request by SkyKilla**
Added: 
 copySheet()
 _rebuild()
Modified: 
 deleteSheet()

### CODE REVIEW

**Comment by kant2002** [comment]
Could you rebase your PR against master? Seems to be this PR based on older version of XLSX-template, so it hard for me understand what from your change is required, and what is crust.

I seems to get idea why this PR is needed, would be extremely cool if you split this PR on separate parts since it will be easier to review and create tests for functionality which you implement.

For example since you have
- copySheet
- deleteSheet
- formula Substitution
I have to create some test templates which will prove correctness of your changes. Also if I have questions regarding some part of functionality, only it will be temporary blocked, and other functionalty will land faster.

I know that it long till this PR floating around, and I was not so responsive as I should.

**Comment by SkyKilla** [comment]
Okay! Rebased against master. 
Work with formulas in next PR.

**Comment by SkyKilla** [comment]
"Thousands of apologies" :)
Rebase with 
"git remote add upstream git://github.com/optilude/xlsx-template.git"
"git pull upstream master"

Add my changes after that,..Think, I fix it now.

File: lib/index.js [hunk 1]
```diff
@@ -57,32 +57,85 @@ module.exports = (function() {
...
 
-      var id = self.sheets.findIndex(function(el){return el.id == sheet.id});
-      self.sheets.splice(id, 1);
+    /**
+    * Clone sheets in current workbook template
+    */
+    Workbook.prototype.copySheet = function(sheetName, copyName){
+      var self = this;
+      var sheet = self.loadSheet(sheetName); //filename, name , id, root
+      var indexS = (self.workbook.findall("sheets/sheet").length+1).toString();
```
Comment (kant2002) [inline lib/index.js]: Would be good to rename variable to `nextSheetIndex`, or `newSheetIndex`.

File: lib/index.js [hunk 2]
```diff
@@ -57,32 +57,85 @@ module.exports = (function() {
...
 
-      var el = self.workbook.find("definedNames");
-      if (el){
-        el.delItem(id);
-        el.findall("definedName").forEach(function(item, index){
-          item.attrib.localSheetId = index.toString()
-        })
-      }
+    /**
+    *  Partially rebuild nodes after copy/delete sheets
```
Comment (kant2002) [inline lib/index.js]: Could you explain a bit more what's purpose of `_rebuild` function. 
Looks like it make indexes of elements sequential, but why this is needed? is this Excel requirements, or purely aesthetical. That allow doing safe changes in that function later one.

File: lib/index.js (hunk 1 above)
Comment (SkyKilla) [reply lib/index.js]: Done

File: lib/index.js (hunk 2 above)
Comment (SkyKilla) [reply lib/index.js]: each <sheet> 'r:id' attribute in '\xl\workbook.xml'
must point to correct <Relationship> 'Id' in xl\_rels\workbook.xml.rels
copy an delete break this rule
code simplified and comments added in last commit. I dont like name "_rebuild".
maybe "_restoreRels"  || "_restoreRelationshipIDs" is more convenient

**Pull request merged**